File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        ans = 0
        
        for num in nums:
            if num - 1 not in s:
                nn = num
                curlen = 1
                while nn + 1 in s:
                    curlen += 1
                    nn += 1
                
                ans = max(ans, curlen)
        
        return ans

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s;
        
        for(int num: nums) s.insert(num);
        
        int ans = 0;
        
        for (int num: nums) {
            if (s.find(num - 1) == s.end()){
                int curnum = num;
                int curlen = 1;
                
                while (s.find(curnum + 1) != s.end()){
                    curlen++;
                    curnum++;
                }
                
                if (curlen > ans) ans = curlen;
            }
        }
        
        return ans;
    }
};